### PR TITLE
chore: airdrops general improvements

### LINKED ISSUE #1641: Airdrops general improvements
- improve gas estimations
- fix gas pumping
- check for legislator badges before giving them away

### PR DESCRIPTION
**Pull request by 1emu**
This PR adds a new secret env `BLOCKNATIVE_API_KEY`

Closes #1641 

Estimations seem to work better (faster txs)
image

Gas increment is properly calculated
image

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ If you are running this project locally you only need to check the following env
 - `COMMITTEE_ADDRESSES`: list of eth addresses separated by `,` that will be able to enact finished proposals
 - `DISCOURSE_API_KEY`: the api key use to publish the proposals on the forum
 - `RPC_PROVIDER_URL`: the rpc provider to get the latest block
+- `BLOCKNATIVE_API_KEY`: api key for [blocknative](https://www.blocknative.com/), a polygon tx gas estimation service
```
Comment (andyesp) [inline README.md:51]: Maybe add why are we using this service in the project?

File: src/constants.ts [hunk 2]
```diff
@@ -14,6 +13,14 @@ function getBooleanStringVar(variableName: string, defaultValue: boolean) {
   return defaultValue
 }
 
+export function trimOtterspaceId(rawId: string) {
```
Comment (andyesp) [inline src/constants.ts:16]: Why is this function in `constants`? 🤔

File: src/clients/BlockNative.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      const blockPrice = data.blockPrices[0] // current block
+      const estimatedPrice = blockPrice.estimatedPrices[0] // highest confidence estimation
+
+      return {
+        maxFeePerGas: ethers.utils.parseUnits(estimatedPrice.maxFeePerGas.toString(), 'gwei'),
+        maxPriorityFeePerGas: ethers.utils.parseUnits(estimatedPrice.maxPriorityFeePerGas.toString(), 'gwei'),
+      }
+    } catch (error) {
+      ErrorService.report('Error while fetching polygon gas prices', { error, category: ErrorCategory.Badges })
+      return {
```
Comment (andyesp) [inline src/clients/BlockNative.ts:30]: Should we continue even if the estimation fails?

**Review by andyesp — APPROVED** [review]

File: src/back/utils/contractInteractions.ts [hunk 4]
```diff
@@ -20,35 +24,52 @@ function getBadgesSignerAndContract() {
...
+  contract: Contract,
+  formattedRecipients: string[],
+  ipfsAddress: string,
+  gasIncrement: number
+) {
+  const defaultGasConfig = await estimateGas(async () => contract.estimateGas.airdrop(formattedRecipients, ipfsAddress))
+  return gasIncrement > 0 ? increaseGas(defaultGasConfig, gasIncrement) : defaultGasConfig
+}
+
+function increaseGas(gasConfig: GasConfig, gasIncrement: number) {
```
Comment (andyesp) [inline src/back/utils/contractInteractions.ts:63]: `getIncreasedGasConfig` or `getGasConfigIncreased`?

File: src/constants.ts (hunk 2 above)
Comment (1emu) [reply src/constants.ts:10]: Yes, I tried moving it but with no luck

**Pull request merged**